### PR TITLE
Add close button option

### PR DESCRIPTION
**Pull request by Unforgiven-wanda**
- Updated material-ui@next and fixed some issues issues
- Added close button option (Added material-ui-icons peerDependency)
- Undo message customization (You can now change the text of 'Undo'. Useful if english isn't the intended language)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,7 +14,8 @@
   },
   "repository": {
     "type": "git",
-    "url": "git+https://github.com/TeamWertarbyte/material-ui-snackbar-provider.git"
+    "url":
```
Comment (leMaik) [inline package.json]: No need to change line breaking here. Or was it done automatically?

File: package.json [hunk 2]
```diff
@@ -27,9 +28,11 @@
   "author": "Wertarbyte <kontakt@wertarbyte.com> (https://wertarbyte.com)",
   "license": "MIT",
   "bugs": {
-    "url": "https://github.com/TeamWertarbyte/material-ui-snackbar-provider/issues"
+    "url":
+      "https://github.com/TeamWertarbyte/material-ui-snackbar-provider/issues"
```
Comment (leMaik) [inline package.json]: See above

File: package.json [hunk 3]
```diff
@@ -27,9 +28,11 @@
...
   },
-  "homepage": "https://github.com/TeamWertarbyte/material-ui-snackbar-provider#readme",
+  "homepage":
+    "https://github.com/TeamWertarbyte/material-ui-snackbar-provider#readme",
```
Comment (leMaik) [inline package.json]: See above

File: README.md [hunk 4]
```diff
@@ -59,6 +59,7 @@ export default withSnackbar()(MyComponent) // export the wrapped component
 * `message` (string) – message to display
 * `action` (string, _optional_) – label for the action button
 * `handler` (function, _optional_) – click handler for the action button
+* `close` (boolean, _optional_) – Whether to show the close button or not
```
Comment (leMaik) [inline README.md]: :thinking: I think we'll be more future-proof if we made the fourth argument an `options` object, with a `closeable` option.

File: src/SnackbarProvider.js [hunk 5]
```diff
@@ -27,56 +30,76 @@ export default class SnackbarProvider extends PureComponent {
...
+    let actions = [];
+    if (action != null)
+      actions.push(
+        <Button
+          key="button"
+          color="secondary"
+          size="small"
+          onClick={this.handleActionClick}
+        >
+          {" "}
```
Comment (leMaik) [inline src/SnackbarProvider.js]: What is this supposed to do?

**Comment by Unforgiven-wanda** [comment]
Sure, I'll fix it as soon as I can.

**Pull request closed without merging**